sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/core/lang/SqlDelightFile.kt

```diff
@@ -38,6 +39,15 @@ class SqlDelightFile(
     viewProvider: FileViewProvider
 ) : SqliteFileBase(viewProvider, SqlDelightLanguage),
     SqliteAnnotatedElement {
+
+  /**
+   * Hack: We cannot pass in QueryIdGenerator through constructor
+   * as the current Text Fixture framework breaks.
+   * It seems the ParserDefinition registered to mock IntelliJ environment does not get reset between tests
+   * and when the SqlDelightFile is insntiated it re-uses previous test's QueryIdGenerator
+   */
+  var queryIdGenerator: QueryIdGenerator? = null
```

Comment:
this is the bit I'm most cautious about. Do you mind going a bit more in to what the original attempt was and what problem came up?

QueryIdGenerator is meant to be one-per-database right? One per task invocation? Understanding that will help me better understand how to model it for the intellij plugin.

Reply:
Currently, `QueryIdGenerator` is instantiated per `generate*Interface` task or in other words per-database. 

The hack here is that I attach an instance of `QueryIdGenerator` to `SqlDelightFile` after its created instead of passing into the constructor [here](https://github.com/square/sqldelight/blob/master/sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/core/lang/SqlDelightParserDefinition.kt#L58).  The reason for doing this is that the `SqlDelightParserDefinition` registered to the mock IntelliJ environment does not seem to reset when calling `FixtureCompiler#parseSql`. This in turn leads to the counter not being reset across unit tests.

Fixing the test fixture to support passing `QueryIdGenerator` through `SqlDelightFile` constructor would be the ideal solution here.